modules/sessions/hooks/useSignUp.ts

```diff
@@ -0,0 +1,42 @@
+import { useRouter } from 'next/navigation';
+
+import useApi from 'modules/share/hooks/useApi';
+
+import { SignUpFormPayload, SignUpFormValues } from '../types';
+
+const useSignUp = () => {
+  const router = useRouter();
+
+  const { fetchApi, isLoading } = useApi(
+    'sign-up',
+    (values: SignUpFormPayload) => ({
+      method: 'POST',
+      url: '/api/signup',
+      data: values,
+    }),
+    {
+      revalidate: false,
+      rollbackOnError: false,
+      // TODO: sign-in before redirect
+      // TODO: add notification onError
+      onSuccess: () => router.push('/verify-account'),
```

Comment:
Expected:

1. user input info
2. click submit button
3. call register api
4. implicit sign-in
5. redirect to verify-account page
However, manual sign-in is not ready yet.

Reply:
We need something to notify user that there is an error.
Suggest: react-toasify

Reply:
@pvnamGit we need to set up a toast or something like that, but it will be done in another task